src/main/java/uk/gov/ida/matchingserviceadapter/domain/EncryptedAssertionContainer.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.ida.matchingserviceadapter.domain;
+
+import org.opensaml.saml.saml2.core.AttributeQuery;
+import org.opensaml.saml.saml2.core.EncryptedAssertion;
+import uk.gov.ida.saml.security.validators.ValidatedEncryptedAssertionContainer;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/* Wrapper class to put an attribute query in to allow it to function as a ValidateEncryptedAssertionContainer */
+public class EncryptedAssertionContainer implements ValidatedEncryptedAssertionContainer {
+    private final AttributeQuery attributeQuery;
+
+    public EncryptedAssertionContainer(AttributeQuery attributeQuery) {
+        this.attributeQuery = attributeQuery;
+    }
+
+    @Override
+    public List<EncryptedAssertion> getEncryptedAssertions() {
+        return (List<EncryptedAssertion>) (List<?>)  attributeQuery.getSubject()
+                .getSubjectConfirmations().stream()
+                .flatMap(
+                        s -> s.getSubjectConfirmationData().getUnknownXMLObjects(EncryptedAssertion.DEFAULT_ELEMENT_NAME).stream()
```

Comment:
If you cast this stream to `EncryptedAssertion` (`.map(x -> (EncryptedAssertion) x)`) I think you can avoid the double cast above.

Reply:
See if we can cast the Subject into a subtype that contains a `getEncryptedAssertions` method